keptn/keptn
# feat(bridge): Login via OpenID (#6076) (#6077)

Original PR: #6341

Closes #6076
Closes #6077 

## This PR
- adds support to login via a custom OpenID provider

### How to test
The following environment variables have to be set
- `OAUTH_ENABLED`: true
- `OAUTH_BASE_URL`: e.g. http://localhost:3000 or http://myBridgeInstallation.com
- `OAUTH_DISCOVERY`: https://api.login.yahoo.com/.well-known/openid-configuration
- `OAUTH_CLIENT_ID`: myClientID
- `OAUTH_CLIENT_SECRET` (optional): myClientSecret
- `OAUTH_ID_TOKEN_ALG` (optional): myAlg (e.g. `ES256`). Default is `RS256`
- `OAUTH_SCOPE` (optional): Additional scopes that should be added to the authentication flow (e.g. `profile email`), separated by space.
- `OAUTH_NAME_PROPERTY` (optional): The property of the ID token that identifies the user (e.g. `username`). Default is `name`
![image](https://user-images.githubusercontent.com/11599148/145979592-cb5506ab-a908-4fea-9789-cdb79b5ca567.png)


On logout, the user is redirected to a logout page (to prevent being immediately logged in again)
![image](https://user-images.githubusercontent.com/11599148/150313127-8cd97f40-61cd-47d3-b15a-b0a4765a3377.png)


Signed-off-by: Klaus Strießnig <k.striessnig@gmail.com>
